### DISCUSSION TITLE
only validate the current item using trigger

### DISCUSSION THREAD

**Discussion by behzadmehrabi** [react-hook-form/react-hook-form#5895] [Q&A] [answered]
Hi, I have a set of inputs and I want to validate each one of them on their own onBlur,
for that, I use `trigger` and do something like this:
```js
onBlur={() => {
  // name is `items.${currentIndex}.name` 
  trigger(name);
  onBlur();
}}
```

but the validation is called for every field, so if I have 50 items, it will be called for every one of them and not just the current item, how can I fix that?

codesandbox:
https://codesandbox.io/s/beautiful-rubin-2b8vg?file=/src/App.tsx:722-809

**Comment by bluebill1049** [comment, marked as answer]
as far as I know, schema validation will require the entire form values to be validated against the entire schema, there is some library like `Vest` is try to make it more performant by isolated single input validation, but the majority of schema validation will require to validate the whole form.

**Reply by bluebill1049** [reply]
by the way, `trigger` api provided by a name should only re-render based on that input's validation result, and only build in validation can focus on single input validation at a time.

**Reply by behzadmehrabi** [reply]
okay, interesting.
thank you for your answer. :)

**Reply by behzadmehrabi** [reply]
in our application, it's possible to have a  list of 1000 objects, where each object basically has 15 values(each value is an input).
so the whole form validation is gonna take so much time in large-scale data like this.  
I'm gonna try out `vest` and see how it goes :),

maybe I should write the one who has support for partial form validation.

again thank you for your fast response.

**Reply by bluebill1049** [reply]
sorry, I may miss lead you, I think you will have to do the following:

- Build a custom resolver 
- inside resolver only validate inputs which exists in the options (third argument for the callback function) 

I think only Vest support validate single field at a time as schema.

**Reply by behzadmehrabi** [reply]
Thank you so much.
